flat_kivy_extensions/flatappextension.py

```diff
@@ -0,0 +1,366 @@
+        self.theme_manager.add_theme('app', 'navigationdrawer', navigationdrawer)
+
+        self.theme_manager.types_to_theme['CustomIconButton'] = CustomIconButton
+        self.theme_manager.types_to_theme['FlatIconButtonLeft'] = FlatIconButtonLeft
+
+    def setup_font_ramps(self):
+        super(ExtendedFlatApp, self).setup_font_ramps()
+        font_styles = {
+            'HeaderTitle': {
+                'font': 'Roboto-Bold.ttf',
+                # Question: what is the best way to include additional fonts?
+                #'font': 'proximanova-bold-webfont.ttf',
+                'sizings': {'mobile': (25, 'sp'), 'desktop': (20, 'sp')},
+                'alpha': .87,
+                'wrap': False,
+                },
+            'NavigationButton': {
+                'font': 'Roboto-Bold.ttf',
+                # 'font': 'proximanova-bold-webfont.ttf',
+                'sizings': {'mobile': (16, 'sp'), 'desktop': (14, 'sp')},
+                'alpha': .87,
+                'wrap': False,
+            },
+            'NavigationLabelMainHeading': {
+                'font': 'Roboto-Bold.ttf',
+                # 'font': 'proximanova-bold-webfont.ttf',
+                'sizings': {'mobile': (20, 'sp'), 'desktop': (17, 'sp')},
+                'alpha': .87,
+                'wrap': False,
+            },
+            'NavigationLabelSubHeading': {
+                'font': 'Roboto-Bold.ttf',
+                # 'font': 'proximanova-bold-webfont.ttf',
+                'sizings': {'mobile': (18, 'sp'), 'desktop': (15, 'sp')},
+                'alpha': .87,
+                'wrap': False,
+            },
+            }
+
+        for each in font_styles:
+            style = font_styles[each]
+            sizings = style['sizings']
+            style_manager.add_style(style['font'], each, sizings['mobile'],
+                sizings['desktop'], style['alpha'])
+
+    def _create_navigation_label_from_string(self, entry):
+        entry = {'text' : entry, 'theme' : ('app', 'navigationdrawer')}
+        if self._first_navigation_label is None:
+            self._first_navigation_label = self._create_navigation_label_from_dict(entry)
+            return
+        entry['style'] = 'NavigationLabelSubHeading'
+        self._create_navigation_label_from_dict(entry)
+
+    def _create_navigation_label_from_dict(self, entry):
+        label = FlatLabel(text=entry.get('text', 'None'))
+        if 'theme' in entry.keys():
+            setattr(label, 'theme', entry['theme'])
+            del(entry['theme'])
+        else:
+            setattr(label, 'theme', ('app', 'navigationdrawer'))
+        for (key,value) in entry.items():
+            setattr(label, key, value)
+        self._side_panel.add_widget(label)
+        return label
+
+    def _create_navigation_button(self, entry):
+        btnText = entry[0]
+        btn = FlatIconButtonLeft(text=btnText)
+        btn.theme = ('app', 'navigationdrawer')
+
```

Comment:
Either or is ok to do this. Another way to do it would be to have btn pass its theme to the icon and label children and then style them with this type of static declaration sin the theme as well